astropy/nddata/nduncertainty.py

```diff
@@ -137,14 +137,18 @@ def unit(self):
 
         If the unit is not set the unit of the parent will be returned.
         """
-        if self._unit is None:
-            if (self._parent_nddata is None or
-                    self.parent_nddata.unit is None):
-                return None
-            else:
-                return self.parent_nddata.unit
         return self._unit
 
+    @unit.setter
+    def unit(self, value):
+        """Each subclass should interpret how it transform parent's unit into the correct Uncertainty Unit (
```

Comment:
Please remove parenthesis at the end of this line

Reply:
of course :)